Address puma issues

imtayadeway opened the pull request:
- [x] ManageIQ#3477
- [x] ManageIQ#3468
- [x] ManageIQ#3444
- [x] Find out why the connection pool had to be increased, and if this change was necessary

matthewd commented on Gemfile:
```diff
@@ -41,7 +41,7 @@ gem "acts_as_tree",                   "~>2.1.0"  # acts_as_tree needs to be requ
 gem "american_date"
 gem "default_value_for",              "~>3.0.1"
-gem "thin",                           "~>1.3.1"  # Used by rails server through rack
+gem "puma",                           "!=2.12.0"
```
As 2.12.1 is out, we probably needn't bother with this

imtayadeway commented:
Just rebased, squashed the now-redundant puma version changes in the Gemfile. Switched to puma master to incorporate @matthewd 's bugfix - don't know if we want to leave it like that for now or wait until v.2.12.3 before this gets merged. I also incorporated @chessbyte 's suggestion to keep the default server as thin for now, but allow it to be configured through `config/vmdb.yml`. 

AFAIK this should resolve all open issues related to puma.

matthewd commented on lib/workers/mixins/web_server_worker_mixin.rb:
```diff
@@ -11,7 +11,7 @@ def do_work
     @worker.release_db_connection
 
-    # Since thin traps interrupts, log that we're going away and update our worker row
+    # Since puma traps interrupts, log that we're going away and update our worker row
```
Should we word this more generically, or just as `thin/puma`?

matthewd commented on lib/miq_environment.rb:
```diff
@@ -4,7 +4,7 @@ module MiqEnvironment
     def self.is_rails_server?
       return @is_rails_server unless @is_rails_server.nil?
-      return @is_rails_server = File.basename($PROGRAM_NAME) =~ /rackup|thin/ || (File.basename($PROGRAM_NAME) == "rails" && !!defined?(Rails::Server))
+      return @is_rails_server = File.basename($PROGRAM_NAME) =~ /rackup|puma/ || (File.basename($PROGRAM_NAME) == "rails" && !!defined?(Rails::Server))
```
Should match both

jrafanie replied:
Maybe we keep thin and puma?

jrafanie commented on app/models/mixins/miq_web_server_worker_mixin.rb:
```diff
@@ -13,9 +13,13 @@ def self.binding_address
     end
 
+    def self.rails_server
+      VMDB::Config.new("vmdb").config.fetch_path(:server, :rails_server)
```
```ruby
VMDB::Config.new("vmdb").config.fetch_path(:server, :rails_server) || "thin"
```

jrafanie replied:
(existing dev setups won't have this value in the yaml file)

jrafanie commented on Gemfile:
```diff
@@ -42,6 +42,7 @@ gem "acts_as_tree",                   "~>2.1.0"  # acts_as_tree needs to be requ
 gem "american_date"
 gem "default_value_for",              "~>3.0.1"
 gem "thin",                           "~>1.3.1"  # Used by rails server through rack
+gem "puma",                                                              :git => "git://github.com/puma/puma.git", :branch => "master"
```
thanks @matthewd  :+1:   :heart:

jrafanie commented:
@imtayadeway Can you rewrite the commits to use the `Fixes #1234` syntax?  If you use that syntax, those bugs will be closed automatically when the commit gets merged.

imtayadeway commented:
@jrafanie @matthewd thanks for catching those! I just rebased and reworded. Also changed the order of things a bit to (hopefully) make it a bit more coherent. Can you re-review?

jrafanie commented:
@imtayadeway Looks good although... does `addresses` also auto-close issues on merge?  Two of the commits have `addresses` in message.  I'm only aware of these:  https://help.github.com/articles/closing-issues-via-commit-messages/

imtayadeway commented:
@jrafanie damn! I looked through that list while working on the bot only last week....Don't know how I made that error! Let me just, uh, address that :wink:

jrafanie commented:
> Don't know how I made that error! Let me just, uh, address that

:laughing:

jrafanie commented:
LGTM @imtayadeway, please remove the [WIP] from the title when it's ready for final review/merge.

imtayadeway commented:
@jrafanie done :grin:

gtanzillo commented:
:+1:
